org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/AbstractScanTest.java

```diff
@@ -473,7 +475,8 @@ private void checkRun(IRunnableDevice<ScanModel> scanner) throws Exception {
 		smodel.setPositionIterable(gen);
 		smodel.setDetectors(detector);
 		smodel.setBean(bean);
-		if (monitor!=null) smodel.setMonitors(monitor);
+		if (monitorsPerPoint!=null) smodel.setMonitorsPerPoint(monitorsPerPoint);
+		if (monitorsPerScan!=null) smodel.setMonitorsPerScan(monitorsPerScan);
 
 		// Create a scan and run it without publishing events
 		IRunnableDevice<ScanModel> scanner = dservice.createRunnableDevice(smodel, publisher);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "scanner". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)